rust/bindgen/src/imports.rs

```diff
@@ -19,6 +19,28 @@ impl PyImports {
         push(&mut self.typing_imports, module, name.into())
     }
 
+    pub fn merge(&mut self, imports: PyImports) {
```

Comment:
I originally tried to avoid adding this method because imports are automatically renamed if they conflict with things and the merging process doesn't work well where something that wasn't renamed could be renamed otherwise during generation to make things work. 

Would it be possible to swap the `Source` objects perhaps during the generation of the root functions where the "root `Source`" temporarily becomes the generator `Source`? I forget if I've done that here in this bindings generator or only in others.